clients/da-vinci-client/src/main/java/com/linkedin/davinci/helix/LeaderFollowerPartitionStateModel.java

```diff
@@ -196,23 +197,33 @@ public void onBecomeOfflineFromError(Message message, NotificationContext contex
     logger.warn("unexpected state transition from ERROR to OFFLINE");
   }
 
-  void updateLagMonitor(String resourceName, BiConsumer<Version, Integer> lagMonFunction) {
+  void updateLagMonitor(String resourceName, BiConsumer<Version, Integer> lagMonFunction, boolean isNullVersionValid) {
     try {
       String storeName = Version.parseStoreFromKafkaTopicName(resourceName);
       int storeVersion = Version.parseVersionFromKafkaTopicName(resourceName);
       Pair<Store, Version> res = getStoreRepo()
           .waitVersion(storeName, storeVersion, getStoreAndServerConfigs().getServerMaxWaitForVersionInfo(), 200);
       Store store = res.getFirst();
       Version version = res.getSecond();
-      if (store == null || version == null) {
+      if (store == null) {
         logger.error(
-            "Failed to get store or version for resource: {}-{}. store: {} version: {}. Will not update lag monitor.",
+            "Failed to get store for resource: {}-{}. Will not update lag monitor.",
             resourceName,
-            getPartition(),
-            store,
-            version);
+            getPartition());
         return;
       }
+      if (version == null) {
+        if (isNullVersionValid) {
+          // During version deletion, the version will be deleted from ZK prior to servers perform resource deletion.
+          // It's valid to have null version when trying to remove lag monitor for the deleted resource.
+          version = new VersionImpl(storeName, storeVersion, "");
+        } else {
+          logger.error(
```

Comment:
We need `return` here otherwise we'll get NPE. 

Maybe we can restructure this as below?
```java
if (version == null && !isNullVersionValid) {
    logger.error("Failed to get version for resource: {}-{}. Will not update lag monitor.", resourceName, getPartition());
    return;
}

if (version == null) {
    // During version deletion, the version will be deleted from ZK prior to servers perform resource deletion.
    // It's valid to have null version when trying to remove lag monitor for the deleted resource.
    version = new VersionImpl(storeName, storeVersion, "");
}
```

Reply:
Good catch